core/deployment/src/main/java/io/quarkus/deployment/util/GenericTypeUtil.java

```diff
@@ -0,0 +1,160 @@
+package io.quarkus.deployment.util;
+
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.ClassType;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.jboss.jandex.ParameterizedType;
+import org.jboss.jandex.Type;
+import org.jboss.jandex.TypeVariable;
+
+public final class GenericTypeUtil {
```

Comment:
I think I would have made it a `JandexUtil`.

Reply:
No strong opinion on this, so let's go with `JandexUtil`